R/ictr_score_companies.R

```diff
@@ -73,5 +73,25 @@ ictr_score_companies <- function(ecoinvent_scores, companies) {
   ictr_output <- ictr_output |>
     replace(is.na(ictr_output), 0)
 
+  ## replace 0s with NAs in share columns if there is no product or input for
+  ## any company.
+  summed <- ictr_output |>
+    group_by(company_id) |>
+    summarize(across(starts_with("share_"), sum))
+
+  ids_for_na <- summed$company_id[(summed$share_all == 0)
+                                  & (summed$share_unit == 0)
+                                  & (summed$share_sector == 0)
+                                  & (summed$share_unit_sec == 0)]
+
```

Comment:
@maurolepore 
I have applied checks for all the four columns `share_all`, `share_unit`, `share_sector`, `share_unit_sec`. The values in all these four columns come from the `input_co2` column of the `input` table. Hence, if the `share_all` column has sum of `0`, then I don't think any other column will have a value greater than 0. Hence, I would like to update my current code only based on `share_all` column. Please let me know if you don't approve of this! Thanks!

Reply:
Thanks! With this as a guide I did a bunch of refactorings to reduce duplication. You may the the final result in the "Files changed".